### PR TITLE
Refactor refdiff

### PR DESCRIPTION
**Pull request by warren830**
# Summary

refactor refdiff to adopt new design

### Description
Describe what this PR does, and aims to solve in a few sentences.

### Does this close any open issues?
Please mention the issues here.

### Current Behavior
Describe the current behaviour of this issue, if relevant.

### New Behavior
Describe the new behaviour updated in this issue, if relevant.

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Comment by abeizn** [comment]
LGTM

File: plugins/refdiff/refdiff.go [hunk 1]
```diff
@@ -1,136 +1,87 @@
...
-	/* TODO: adopt new method
-	var err error
+	refdiffCmd := &cobra.Command{Use: "refdiff"}
 
-	args := os.Args[1:]
-	if len(args) < 2 {
-		panic(fmt.Errorf("Usage: refdiff <repo_id> <new_ref_name> <old_ref_name>"))
-	}
-	repoId, newRefName, oldRefName := args[0], args[1], args[2]
+	repoId := refdiffCmd.Flags().StringP("repoId", "r", "", "repoId")
```
Comment (klesh) [inline plugins/refdiff/refdiff.go:66]: please use dash-case

**Review by abeizn — APPROVED** [review]

File: plugins/refdiff/tasks/refs_commits_diffs_calculator.go [hunk 2]
```diff
@@ -6,12 +6,18 @@ import (
 	"github.com/cayleygraph/cayley"
 	"github.com/cayleygraph/quad"
 	"github.com/merico-dev/lake/logger"
-	"github.com/merico-dev/lake/models"
 	"github.com/merico-dev/lake/models/domainlayer/code"
 	"github.com/merico-dev/lake/plugins/core"
 	"gorm.io/gorm/clause"
 )
 
+var CalculateCommitsDiffMeta = core.SubTaskMeta{
```
Comment (klesh) [inline plugins/refdiff/tasks/refs_commits_diffs_calculator.go:14]: can we this to the end of file?

File: plugins/refdiff/refdiff.go (hunk 1 above)
Comment (warren830) [reply plugins/refdiff/refdiff.go:66]: fixed

File: plugins/refdiff/tasks/refs_commits_diffs_calculator.go (hunk 2 above)
Comment (warren830) [reply plugins/refdiff/tasks/refs_commits_diffs_calculator.go:14]: ok, fixed

**Review by klesh — APPROVED** [review]
LGTM

**Pull request merged**